# <b> Boilerplate demographic data analyzer with Pandas </b>

<b> Demographic analysis is important because it gives useful information that can be used to make good decisions in business, government, and social services, among other places. It helps people understand the characteristics of a population and how it might change in the future, which is important to make a decisions for business </b>

Data was used to analyzed and reveal the the informations of income of working people across countries and its contributing factors: race, degree, sex and age.

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("adult.data.csv")

Have  look of what informations data file contains :

In [3]:
df.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [4]:
df.shape

(32561, 15)

Data file contains 32561 entries ( rows ) and 15 attributes ( columns)

##### 1. How many people of each race are represented in this dataset?

In [5]:
df.value_counts("race")

race
White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
Name: count, dtype: int64

In this dataset, mostly people are white, So the the information of white people are more true to life, especially in America , than other countries.

##### 2. What is the average age of men?

In [7]:
mean_age = round(df.loc[df["sex"] == "Male"]["age"].mean(),1)
median_age =round(df.loc[df["sex"] == "Male"]["age"].median(),1)
print("Average age is:",mean_age)
print("Median age is:",median_age)

Average age is: 39.4
Median age is: 38.0


<i> The average age of men in this survey is 39,4. <br>
The median age is 38.0. <br>
As we can see the data was collect from people in the working age.

##### 3. What is the percentage of people who have a Bachelor's degree?

In [11]:
bachelor_percentage = df[df["education"] == "Bachelors"]["education"].count() / df["education"].count() * 100
print("The percentage of people who have Bachelor's degree is: {:.2f} %".format(bachelor_percentage))

The percentage of people who have Bachelor's degree is: 16.45 %


<i> As we can see, the percentage of people who have bachelor's degree is 16.45%

##### 4. What percentage of people with advanced education (Bachelors, Masters, or Doctorate) make more than 50K?

 - What the percentage of people with higher education:

In [12]:
higher_education = df[(df["education"] == "Bachelors") | (df["education"] == "Masters") | (df["education"] == "Doctorate")]["education"].count()

- The percentage of people with advanced education  make more than 50K :

In [14]:
higher_education_rich = round(df[((df["education"] == "Bachelors") | (df["education"] == "Masters") | (df["education"] == "Doctorate")) & (df["salary"] == ">50K")]["salary"].count() / higher_education * 100,1)
print("The percentage of people who have advanced education make more than 50k is: {:.2f} %".format(higher_education_rich))

The percentage of people who have advanced education make more than 50k is: 46.50 %


<i> As the dataset, about 1/2 of people who have advanced education make more than 50k per year.

##### 5. What percentage of people without advanced education make more than 50K?

 - What the percentage of people with lower education:

In [16]:
lower_education = df[(df["education"] != "Bachelors") & (df["education"] != "Masters") & (df["education"] != "Doctorate")]["education"].count()

- The percentage of people with advanced education  make more than 50K :

In [17]:
lower_education_rich = round(df[(df["education"] != "Bachelors") & (df["education"] != "Masters") & (df["education"] != "Doctorate")& (df["salary"] == ">50K")]["salary"].count() / lower_education * 100,1)
print("The percentage of people who have advanced lower make more than 50k is: {:.2f} %".format(lower_education_rich))

The percentage of people who have advanced lower make more than 50k is: 17.40 %


<i> Only 17.40% people who have lower education ear more than 50k/ year.

##### 6. What percentage of the people who work the minimum number of hours per week have a salary of more than 50K?

- What is the minimum number of hours a person works per week?

In [18]:
hour_min = df["hours-per-week"].min()

- What percentage of the people who work the minimum number of hours per week have a salary of more than 50K

In [19]:
min_hour_rich = df[(df["hours-per-week"] == df["hours-per-week"].min()) & (df["salary"] == ">50K")]["salary"].count() 
print("The percentage of people who work the minimum number of hours have salara more than 50k is: {:.2f} %".format(min_hour_rich))

The percentage of people who work the minimum number of hours have salara more than 50k is: 2.00 %


<i> 2% of people who work minium hour and earn more than 50k per year

##### 7. What country has the highest percentage of people that earn >50K and what is that percentage?

- What country has the highest number of people that earn >50K?

In [20]:
highest_earning_country = (df[df["salary"] == ">50K"].value_counts("native-country") / df["native-country"].value_counts()).sort_values(ascending=False).index[0]

- What country has the highest percentage of people that earn >50K and what is that percentage?

In [21]:
highest_earning_country_percentage = round((df[df["salary"] == ">50K"].value_counts("native-country") / df["native-country"].value_counts()).sort_values(ascending=False).iloc[0] * 100,1)

In [25]:
print("{} is the country which has the highest of people earn >50k and the percentage of people earn >50k is {:,.2f} %".format(highest_earning_country,highest_earning_country_percentage))

Iran is the country which has the highest of people earn >50k and the percentage of people earn >50k is 41.90 %


<i> People in Iran tend to earn money than other countries.

##### 8. Identify the most popular occupation for those who earn >50K in India.


In [30]:
top_IN_occupation = df[(df["salary"] == ">50K") & (df["native-country"] =="India")].value_counts("occupation").head(1).index[0]
print("{} is the most popular job which earn >50k per year in India".format(top_IN_occupation))

Prof-specialty is the most popular job which earn >50k per year in India


<i> Professor specialty in India is most pooular job which have salara  more than 50k/ year